source/server/http/admin.cc

```diff
@@ -594,10 +596,14 @@ Http::Code AdminImpl::handlerPrometheusStats(absl::string_view, Http::HeaderMap&
 }
 
 std::string PrometheusStatsFormatter::sanitizeName(const std::string& name) {
-  std::string stats_name = name;
-  std::replace(stats_name.begin(), stats_name.end(), '.', '_');
-  std::replace(stats_name.begin(), stats_name.end(), '-', '_');
-  return stats_name;
+  // The name must match the regex [a-zA-Z_][a-zA-Z0-9_]* as required by
+  // prometheus. Refer to https://prometheus.io/docs/concepts/data_model/.
+  std::string stats_name = std::regex_replace(name, PromRegex, "_");
+  if (stats_name[0] >= '0' && stats_name[0] <= '9') {
```

Comment:
I think this case is unneeded, because this is always called from `PrometheusStatsFormatter::metricName()`, which prepends "envoy_" to the metric name.

Reply:
Yes. The check is not necessary for current use case. I will remove the check.

Reply:
@ggreenway ？

Reply:
Good point. I agree it's more reasonable to sanitize the metric and tag name as a whole.